src/network/vlan.rs

```diff
@@ -181,12 +181,19 @@ impl driver::NetworkDriver for Vlan<'_> {
         // a dhcp lease.  it will also perform the IP address assignment
         // to the macvlan interface.
         let subnets = if data.ipam.dhcp_enabled {
-            get_dhcp_lease(
+            let (subnets, dns_servers, domain_name) = get_dhcp_lease(
                 &data.host_interface_name,
                 &data.container_interface_name,
                 self.info.netns_path,
                 &container_vlan_mac,
-            )?
+            )?;
+            if dns_servers.is_some() {
+                response.dns_server_ips = dns_servers;
+            }
```

Comment:
you do not need to check dns_servers.is_some() but you check that self.info.container_dns_servers is empty because these are set via podman run --dns and dhcp should not overwrite them.